app/src/split/bluetooth/central.c

```diff
@@ -584,6 +591,31 @@ int zmk_split_bt_invoke_behavior(uint8_t source, struct zmk_behavior_binding *bi
     return split_bt_invoke_behavior_payload(wrapper);
 }
 
+static zmk_leds_flags_t led_flags = 0;
+
+static void split_central_update_leds_callback(struct k_work *work) {
+    for (int i = 0; i < ZMK_BLE_SPLIT_PERIPHERAL_COUNT; i++) {
+        if (peripherals[i].state != PERIPHERAL_SLOT_STATE_CONNECTED) {
+            continue;
+        }
+
+        int err =
+            bt_gatt_write_without_response(peripherals[i].conn, peripherals[i].update_leds_handle,
+                                           &led_flags, sizeof(led_flags), true);
+
+        if (err) {
+            LOG_ERR("Failed to write the leds characteristic (err %d)", err);
+        }
+    }
+}
+
+static K_WORK_DEFINE(split_central_update_leds, split_central_update_leds_callback);
+
+int zmk_split_bt_update_leds(zmk_leds_flags_t leds) {
+    led_flags = leds;
+    return k_work_submit_to_queue(&split_central_split_run_q, &split_central_update_leds);
```

Comment:
I added a variable at the start of `split_central_update_leds_callback` to avoid reading the variable at each iteration of the loop. Correct me if I'm wrong, but as in the other comment, I think it shouldn't be necessary to use a lock or an atomic variable.